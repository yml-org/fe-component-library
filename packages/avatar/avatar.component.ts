import { AvatarComponent } from './src/avatar/avatar.component';

declare global {
  interface HTMLElementTagNameMap {
    'ymlwebcl-avatar': AvatarComponent;
  }
}
