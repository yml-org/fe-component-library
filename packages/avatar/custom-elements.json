{
  "schemaVersion": "1.0.0",
  "readme": "",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "avatar.component.ts",
      "declarations": [],
      "exports": []
    },
    {
      "kind": "javascript-module",
      "path": "src/avatar/avatar.component.ts",
      "declarations": [
        {
          "kind": "class",
          "description": "",
          "name": "AvatarComponent",
          "members": [
            {
              "kind": "field",
              "name": "isRounded",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "attribute": "isRounded",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "width",
              "type": {
                "text": "number | undefined"
              },
              "default": "75",
              "attribute": "width",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "height",
              "type": {
                "text": "number | undefined"
              },
              "default": "75",
              "attribute": "height",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "usePercentage",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "attribute": "usePercentage",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "hasBorder",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "attribute": "hasBorder",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "borderColor",
              "type": {
                "text": "string | undefined"
              },
              "default": "'slate-200'",
              "attribute": "borderColor",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "defaultBgColor",
              "type": {
                "text": "string | undefined"
              },
              "attribute": "defaultBgColor",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "slotName",
              "type": {
                "text": "string | undefined"
              },
              "attribute": "slotName",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "onAvatarClick",
              "type": {
                "text": "() => void | undefined"
              },
              "attribute": "onAvatarClick",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "hasShadow",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "attribute": "hasShadow",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "showDefaultIcon",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "attribute": "showDefaultIcon",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "defaultImage",
              "type": {
                "text": "string | undefined"
              },
              "attribute": "defaultImage",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "avatarPartAttribute",
              "type": {
                "text": "string | undefined"
              },
              "default": "'webcl-avatar'",
              "attribute": "avatarPartAttribute"
            },
            {
              "kind": "method",
              "name": "getContainerClassList",
              "privacy": "private"
            },
            {
              "kind": "method",
              "name": "getContainerDimensions",
              "privacy": "private"
            },
            {
              "kind": "method",
              "name": "renderDefaultIcon",
              "privacy": "private"
            }
          ],
          "attributes": [
            {
              "name": "isRounded",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "fieldName": "isRounded"
            },
            {
              "name": "width",
              "type": {
                "text": "number | undefined"
              },
              "default": "75",
              "fieldName": "width"
            },
            {
              "name": "height",
              "type": {
                "text": "number | undefined"
              },
              "default": "75",
              "fieldName": "height"
            },
            {
              "name": "usePercentage",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "fieldName": "usePercentage"
            },
            {
              "name": "hasBorder",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "fieldName": "hasBorder"
            },
            {
              "name": "borderColor",
              "type": {
                "text": "string | undefined"
              },
              "default": "'slate-200'",
              "fieldName": "borderColor"
            },
            {
              "name": "defaultBgColor",
              "type": {
                "text": "string | undefined"
              },
              "fieldName": "defaultBgColor"
            },
            {
              "name": "slotName",
              "type": {
                "text": "string | undefined"
              },
              "fieldName": "slotName"
            },
            {
              "name": "onAvatarClick",
              "type": {
                "text": "() => void | undefined"
              },
              "fieldName": "onAvatarClick"
            },
            {
              "name": "hasShadow",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "fieldName": "hasShadow"
            },
            {
              "name": "showDefaultIcon",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "fieldName": "showDefaultIcon"
            },
            {
              "name": "defaultImage",
              "type": {
                "text": "string | undefined"
              },
              "fieldName": "defaultImage"
            },
            {
              "name": "avatarPartAttribute",
              "type": {
                "text": "string | undefined"
              },
              "default": "'webcl-avatar'",
              "fieldName": "avatarPartAttribute"
            }
          ],
          "mixins": [
            {
              "name": "TailwindElement",
              "module": "/src/shared/tailwind.element"
            }
          ],
          "superclass": {
            "module": "src/avatar/avatar.component.ts"
          },
          "tagName": "ymlwebcl-avatar",
          "customElement": true
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "AvatarComponent",
          "declaration": {
            "name": "AvatarComponent",
            "module": "src/avatar/avatar.component.ts"
          }
        },
        {
          "kind": "custom-element-definition",
          "name": "ymlwebcl-avatar",
          "declaration": {
            "name": "AvatarComponent",
            "module": "src/avatar/avatar.component.ts"
          }
        }
      ]
    }
  ]
}
