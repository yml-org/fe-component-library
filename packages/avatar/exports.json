{
  ".": "./src/index.js",
  "./src/*.js": "./src/*.js",
  "./package.json": "./package.json",
  "./avatar.component.js": "./avatar.component.js"
}
