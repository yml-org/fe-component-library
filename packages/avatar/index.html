<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>web components: Avatar</title>
  </head>
  <body>
    <avatar-component></avatar-component>

    <script type="module" src="src/avatar/avatar.component.ts"></script>
  </body>
</html>
