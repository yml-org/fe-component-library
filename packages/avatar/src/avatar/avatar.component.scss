.parent-container {
  avatar-component::part(webcl-avatar) {
    background-color: black;
    color: white;

    span {
      font-weight: 600;
      font-size: 24px;
    }
  }
}