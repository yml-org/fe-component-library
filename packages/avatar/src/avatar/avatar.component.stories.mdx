import { html } from 'lit';
import { Story, Meta, Canvas } from '@storybook/addon-docs';
import './avatar.component.ts';
import {
  roundedVariant,
  withDefaultIcon,
  withCustomHeightWidth,
  withRoundedBorderAvatar,
  withShadow,
  withSlot,
} from '../constants/avatar.variants.ts';
import './avatar.component.scss';

<Meta title="Components/Avatar" />

# Avatar Component

The Avatar component is a Lit component that renders an avatar component that is framework independent and utilizes Tailwind CSS internally.

# Props

| Prop Name       | Type     | Description                                                              |
| --------------- | -------- | ------------------------------------------------------------------------ |
| isRounded       | boolean  | Flag to determine whether or not to have rounded borders                 |
| width           | number   | The width of the avatar component                                        |
| height          | number   | The height of the avatar component                                       |
| usePercentage   | boolean  | Flag to determine whether or not the dimensions should be in percentages |
| hasBorder       | boolean  | Flag to determine whether the border is shown or not                     |
| borderColor     | string   | The border color                                                         |
| slotName        | string   | The name of the child slot                                               |
| onAvatarClick   | Function | The click function that gets triggered on avatar component click         |
| hasShadow       | boolean  | Flag to determine whether the avatar component has box shadow or not     |
| showDefaultIcon | boolean  | Flag to determine whether tho show default avatar icon or not            |
| defaultImage    | string   | The default image to be displayed                                        |
| defaultBgColor  | string   | The default background color provided by the user                        |

# Examples

Avatar component with rounded border with default background

export const roundedBorderTemplate = ({ isRounded, defaultBgColor }) =>
  html`<ymlwebcl-avatar
    .isRounded=${isRounded}
    .defaultBgColor=${defaultBgColor}
  ></ymlwebcl-avatar>`;

<Canvas>
  <Story name="rounded border" args={roundedVariant}>
    {roundedBorderTemplate.bind({})}
  </Story>
</Canvas>

Avatar component with default icon

export const defaultIconTemplate = ({ showDefaultIcon, defaultImage }) =>
  html`<ymlwebcl-avatar
    .showDefaultIcon=${showDefaultIcon}
    .defaultImage=${defaultImage}
  ></ymlwebcl-avatar>`;

<Canvas>
  <Story name="default icon" args={withDefaultIcon}>
    {defaultIconTemplate.bind({})}
  </Story>
</Canvas>

Avatar component with custom width and height

export const withCustomWidthHeightTemplate = ({
  showDefaultIcon,
  defaultImage,
  width,
  height,
}) =>
  html`<ymlwebcl-avatar
    .showDefaultIcon=${showDefaultIcon}
    .defaultImage=${defaultImage}
    .width=${width}
    .height=${height}
  ></ymlwebcl-avatar>`;

<Canvas>
  <Story name="custom height and width" args={withCustomHeightWidth}>
    {withCustomWidthHeightTemplate.bind({})}
  </Story>
</Canvas>

Avatar component with border

export const withRoundedBorderAvatarTemplate = ({
  showDefaultIcon,
  defaultImage,
  isRounded,
  hasBorder,
}) =>
  html`<ymlwebcl-avatar
    .showDefaultIcon=${showDefaultIcon}
    .defaultImage=${defaultImage}
    .isRounded=${isRounded}
    .hasBorder=${hasBorder}
  ></ymlwebcl-avatar>`;

<Canvas>
  <Story
    name="with rounded border avatar variant"
    args={withRoundedBorderAvatar}
  >
    {withRoundedBorderAvatarTemplate.bind({})}
  </Story>
</Canvas>

Avatar component with shadow and border

export const withShadowTemplate = ({
  showDefaultIcon,
  defaultImage,
  isRounded,
  hasBorder,
  hasShadow,
}) =>
  html`<ymlwebcl-avatar
    .showDefaultIcon=${showDefaultIcon}
    .defaultImage=${defaultImage}
    .isRounded=${isRounded}
    .hasBorder=${hasBorder}
    .hasShadow=${hasShadow}
  ></ymlwebcl-avatar>`;

<Canvas>
  <Story name=" avatar with shadow" args={withShadow}>
    {withShadowTemplate.bind({})}
  </Story>
</Canvas>

Avatar component with slot and custom children

export const withSlotAndChildrenTemplate = ({
  isRounded,
  hasBorder,
  hasShadow,
  slotName,
}) =>
  html`<ymlwebcl-avatar
    .slotName=${slotName}
    .isRounded=${isRounded}
    .hasBorder=${hasBorder}
    .hasShadow=${hasShadow}
  >
    <div slot=${slotName}>${'YML'.charAt(0)}</div>
  </ymlwebcl-avatar>`;

<Canvas>
  <Story name="with slot and children" args={withSlot}>
    {withSlotAndChildrenTemplate.bind({})}
  </Story>
</Canvas>

Avatar component with custom styling

export const withCustomStylingTemplate = ({
  isRounded,
  hasBorder,
  hasShadow,
  slotName,
}) =>
  html` <div class="parent-container">
    <ymlwebcl-avatar
      .slotName=${slotName}
      .isRounded=${isRounded}
      .hasBorder=${hasBorder}
      .hasShadow=${hasShadow}
    >
      <div slot=${slotName}>
        <span> ${'YML'.charAt(0)}</span>
      </div>
    </ymlwebcl-avatar>
  </div>`;

<Canvas>
  <Story name="with custom styling" args={withSlot}>
    {withCustomStylingTemplate.bind({})}
  </Story>
</Canvas>
