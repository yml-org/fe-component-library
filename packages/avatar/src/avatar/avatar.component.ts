import { html } from 'lit';
import { customElement, property } from 'lit/decorators.js';
import { TailwindElement } from '../shared/tailwind.element';
import { msg, str } from '@lit/localize';

@customElement('avatar-component')
export class AvatarComponent extends TailwindElement(null) {
  @property()
  isRounded?: boolean = false;
  @property()
  width?: number = 75;
  @property()
  height?: number = 75;
  @property()
  usePercentage?: boolean = false;
  @property()
  slotName?: string;
  @property()
  onAvatarClick?: () => void;
  @property()
  showDefaultIcon?: boolean = false;
  @property()
  defaultImage?: string;
  @property()
  avatarPartAttribute?: string = 'webcl-avatar';

  protected getContainerClassList() {
    return `webcl-avatar-container flex items-center justify-center ${this.isRounded ? 'rounded-full' : ''}`;
  }
  protected getContainerDimensions() {
    const { width, height, usePercentage } = this;
    const dimension = `${usePercentage ? '%' : 'px'}`;
    return `width: ${width}${dimension};height: ${height}${dimension};`;
  }

  protected renderDefaultIcon() {
    const { defaultImage, showDefaultIcon, width, height } = this;
    return showDefaultIcon
      ? html`<img
          src=${defaultImage}
          alt="default icon"
          width="${width}px"
          height="${height}px"
        />`
      : '';
  }

  render() {
    return html`<div
      class=${this.getContainerClassList()}
      style=${this.getContainerDimensions()}
      part=${this.avatarPartAttribute}
      role=${msg(str`button`)}
      @click=${this.onAvatarClick}
    >
      ${this.renderDefaultIcon()}
      ${this.slotName ? html`<slot name=${this.slotName}></slot>` : ''}
    </div>`;
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'avatar-component': AvatarComponent;
  }
}
