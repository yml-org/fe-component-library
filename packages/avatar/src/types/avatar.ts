export type AvatarVariant = {
  isRounded?: boolean;
  width?: number;
  height?: number;
  usePercentage?: boolean;
  slotName?: string;
  onAvatarClick?: () => void;
  showDefaultIcon?: boolean;
  defaultImage?: string;
  hasBorder?: boolean;
  hasShadow?: boolean;
};
