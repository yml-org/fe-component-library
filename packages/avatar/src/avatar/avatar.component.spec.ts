import { AvatarComponent } from './avatar.component';

describe('ymlwebcl-avatar', () => {
  const AVATAR_COMPONENT = 'ymlwebcl-avatar';
  let avatarElement: AvatarComponent;

  const getShadowRoot = (tagName: string): ShadowRoot | null =>
    document.body.getElementsByTagName(tagName)[0].shadowRoot;

  beforeEach(() => {
    avatarElement = window.document.createElement(
      AVATAR_COMPONENT
    ) as AvatarComponent;
    document.body.appendChild(avatarElement);
  });

  afterEach(() => {
    document.body.getElementsByTagName(AVATAR_COMPONENT)[0].remove();
  });

  it('matches component snapshot', async () => {
    expect(
      document.body.getElementsByTagName(AVATAR_COMPONENT)[0]
    ).toMatchSnapshot();
  });

  it('renders the avatar with rounded border', async () => {
    avatarElement['isRounded'] = true;
    await avatarElement.updateComplete;
    const avatarComponentClassList =
      getShadowRoot(AVATAR_COMPONENT)?.querySelector('div')?.classList;
    expect(avatarComponentClassList).toContain('rounded-full');
  });

  it('renders the avatar with custom width', async () => {
    const CUSTOM_WIDTH = 200;
    avatarElement['width'] = CUSTOM_WIDTH;
    await avatarElement.updateComplete;
    const avatarComponentStyle =
      getShadowRoot(AVATAR_COMPONENT)?.querySelector('div')?.style;
    expect(avatarComponentStyle?.width).toBe(`${CUSTOM_WIDTH}px`);
  });

  it('renders the avatar with custom width in percentage', async () => {
    const CUSTOM_WIDTH = 200;
    avatarElement['usePercentage'] = true;
    avatarElement['width'] = CUSTOM_WIDTH;
    await avatarElement.updateComplete;
    const avatarComponentStyle =
      getShadowRoot(AVATAR_COMPONENT)?.querySelector('div')?.style;
    expect(avatarComponentStyle?.width).toBe(`${CUSTOM_WIDTH}%`);
  });

  it('triggers the click event passed', async () => {
    const clickHandler = jest.fn();
    avatarElement['onAvatarClick'] = clickHandler;
    await avatarElement.updateComplete;
    const avatarComponent =
      getShadowRoot(AVATAR_COMPONENT)?.querySelector('div');
    avatarComponent?.click();
    expect(clickHandler).toBeCalled();
  });

  it('renders the component passed in slot', async () => {
    avatarElement['slotName'] = 'avatar';
    window.document.body.appendChild(avatarElement);
    (document.querySelector('ymlwebcl-avatar') || document.body).innerHTML =
      '<div slot="avatar" class="slot-test">Test</div>';
    await avatarElement.updateComplete;
    const elemClassList = document
      ?.querySelector('ymlwebcl-avatar')
      ?.querySelector('div')?.classList;
    expect(elemClassList).toContain('slot-test');
  });
});
