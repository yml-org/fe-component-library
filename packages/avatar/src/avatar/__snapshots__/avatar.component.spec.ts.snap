// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ymlwebcl-avatar matches component snapshot 1`] = `
$ {
  "_$AL": Map {},
  "_$Do": U {
    "_$AA": <!---->,
    "_$AB": null,
    "_$AH": yt {
      "_$AD": O {
        "el": <template />,
        "parts": Array [
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "class",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "style",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "part",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "role",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "click",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "index": 1,
            "type": 2,
          },
          Object {
            "index": 2,
            "type": 2,
          },
        ],
      },
      "_$AM": [Circular],
      "_$AN": undefined,
      "u": Array [
        L {
          "_$AH": "webcl-avatar-container flex items-center justify-center",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="webcl-avatar-container flex items-center justify-center "
            part="webcl-avatar"
            role="button"
            style="width: 75px;height: 75px;"
          >
            
      
            <!--?lit$9078157464$-->
            
      
            <!--?lit$9078157464$-->
            
    
          </div>,
          "name": "class",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        L {
          "_$AH": "width: 75px;height: 75px;",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="webcl-avatar-container flex items-center justify-center "
            part="webcl-avatar"
            role="button"
            style="width: 75px;height: 75px;"
          >
            
      
            <!--?lit$9078157464$-->
            
      
            <!--?lit$9078157464$-->
            
    
          </div>,
          "name": "style",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        L {
          "_$AH": "button",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="webcl-avatar-container flex items-center justify-center "
            part="webcl-avatar"
            role="button"
            style="width: 75px;height: 75px;"
          >
            
      
            <!--?lit$9078157464$-->
            
      
            <!--?lit$9078157464$-->
            
    
          </div>,
          "name": "role",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        Et {
          "_$AH": Symbol(lit-nothing),
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="webcl-avatar-container flex items-center justify-center "
            part="webcl-avatar"
            role="button"
            style="width: 75px;height: 75px;"
          >
            
      
            <!--?lit$9078157464$-->
            
      
            <!--?lit$9078157464$-->
            
    
          </div>,
          "name": "click",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 5,
        },
        U {
          "_$AA": <!--?lit$9078157464$-->,
          "_$AB": 
      ,
          "_$AH": Symbol(lit-nothing),
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
        U {
          "_$AA": <!--?lit$9078157464$-->,
          "_$AB": 
    ,
          "_$AH": Symbol(lit-nothing),
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
      ],
    },
    "_$AM": undefined,
    "_$AN": undefined,
    "_$Cm": true,
    "options": Object {
      "host": [Circular],
      "isConnected": true,
      "renderBefore": null,
    },
    "type": 2,
  },
  "_$EC": undefined,
  "_$E_": Promise {},
  "_$Ei": undefined,
  "_$El": null,
  "__avatarPartAttribute": "webcl-avatar",
  "__height": 75,
  "__isRounded": false,
  "__showDefaultIcon": false,
  "__usePercentage": false,
  "__width": 75,
  "enableUpdating": [Function],
  "hasUpdated": true,
  "isUpdatePending": false,
  "renderOptions": Object {
    "host": [Circular],
    "isConnected": true,
    "renderBefore": null,
  },
  "renderRoot": ShadowRoot {
    "_$litPart$": U {
      "_$AA": <!---->,
      "_$AB": null,
      "_$AH": yt {
        "_$AD": O {
          "el": <template />,
          "parts": Array [
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "class",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "style",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "part",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "role",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "click",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "index": 1,
              "type": 2,
            },
            Object {
              "index": 2,
              "type": 2,
            },
          ],
        },
        "_$AM": [Circular],
        "_$AN": undefined,
        "u": Array [
          L {
            "_$AH": "webcl-avatar-container flex items-center justify-center",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="webcl-avatar-container flex items-center justify-center "
              part="webcl-avatar"
              role="button"
              style="width: 75px;height: 75px;"
            >
              
      
              <!--?lit$9078157464$-->
              
      
              <!--?lit$9078157464$-->
              
    
            </div>,
            "name": "class",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          L {
            "_$AH": "width: 75px;height: 75px;",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="webcl-avatar-container flex items-center justify-center "
              part="webcl-avatar"
              role="button"
              style="width: 75px;height: 75px;"
            >
              
      
              <!--?lit$9078157464$-->
              
      
              <!--?lit$9078157464$-->
              
    
            </div>,
            "name": "style",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          L {
            "_$AH": "button",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="webcl-avatar-container flex items-center justify-center "
              part="webcl-avatar"
              role="button"
              style="width: 75px;height: 75px;"
            >
              
      
              <!--?lit$9078157464$-->
              
      
              <!--?lit$9078157464$-->
              
    
            </div>,
            "name": "role",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          Et {
            "_$AH": Symbol(lit-nothing),
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="webcl-avatar-container flex items-center justify-center "
              part="webcl-avatar"
              role="button"
              style="width: 75px;height: 75px;"
            >
              
      
              <!--?lit$9078157464$-->
              
      
              <!--?lit$9078157464$-->
              
    
            </div>,
            "name": "click",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 5,
          },
          U {
            "_$AA": <!--?lit$9078157464$-->,
            "_$AB": 
      ,
            "_$AH": Symbol(lit-nothing),
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
          U {
            "_$AA": <!--?lit$9078157464$-->,
            "_$AB": 
    ,
            "_$AH": Symbol(lit-nothing),
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
        ],
      },
      "_$AM": undefined,
      "_$AN": undefined,
      "_$Cm": true,
      "options": Object {
        "host": [Circular],
        "isConnected": true,
        "renderBefore": null,
      },
      "type": 2,
    },
  },
}
`;
